src/ios-deploy/ios-deploy.m

```diff
@@ -1714,30 +1729,84 @@ void handle_device(AMDeviceRef device) {
         check_error(AMDeviceStopSession(device));
         check_error(AMDeviceDisconnect(device));
 
-        CFStringRef keys[] = { CFSTR("PackageType") };
-        CFStringRef values[] = { CFSTR("Developer") };
-        CFDictionaryRef options = CFDictionaryCreate(NULL, (const void **)&keys, (const void **)&values, 1, &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks);
-
-        //assert(AMDeviceTransferApplication(afcFd, path, NULL, transfer_callback, NULL) == 0);
-        check_error(AMDeviceSecureTransferPath(0, device, url, options, transfer_callback, 0));
-        close(*afcFd);
-
+        bool incremental_install = app_deltas != NULL;
 
+        CFDictionaryRef options;
```

Comment:
I think it would be better/more-readable to duplicate
```
        AMDeviceConnect(device);
        assert(AMDeviceIsPaired(device));
        check_error(AMDeviceValidatePairing(device));
        check_error(AMDeviceStartSession(device));
```
Into then having these two `if (!incremental_install) ` checks with than in between.

Reply:
Done.